src/Datasource/EntityTrait.php

```diff
@@ -281,8 +299,10 @@ public function &get($property)
 
         if ($this->_methodExists($method)) {
             $result = $this->{$method}($value);
-            return $result;
+            return $recursor($result, $nested);
```

Comment:
Cant you just remove the return statement? Looks like $recursor($result, $nested); will be called the next line anyway then.